src/components/typography/text/text.js

```diff
@@ -45,15 +48,46 @@ const Headline = props => {
 
 Headline.displayName = 'TextHeadline';
 Headline.propTypes = {
-  elementType: PropTypes.oneOf(['h1', 'h2', 'h3']).isRequired,
+  as: requiredIf(
+    PropTypes.oneOf(['h1', 'h2', 'h3']),
+    props => !props.elementType
+  ),
+  elementType(props, propName, componentName, ...rest) {
+    if (props[propName] != null) {
+      const message = oneLine`
+            "${propName}" property of "${componentName}" has been deprecated
+            and will be removed in the next major version.\n
+            Please use "as" prop instead.`;
```

Comment:
We can add a note also to use the codemod for the migration. WDYT?

Reply:
Sure! Although I was thinking of adding that after #747 is merged, because then I can also refactor this to use the `util` that it adds [(this one)](https://github.com/commercetools/ui-kit/blob/4495d051e55a531118f00e5d929cb3a8c75682e6/src/utils/warn-deprecated-prop.js).